src/modal.ts

```diff
@@ -0,0 +1,462 @@
+                    this.hide();
+                    return;
+                } else if (calc < 0)
+                    return;
+                this._wrapper_el.style.right = calc + 'px';
+                if (!this._wrapper_el.style.left)
+                    this._wrapper_el.style.left = this._wrp_init_bounds.x + 'px';
+                this._wrapper_el.style.width = "auto";
+            }
+        }
+        if (ydiff) {
+            let calc;
+            if (this._mouse_pos_mod == "top") {
+                calc = rect.top + ydiff;
+                if ((rect.height - ydiff) < this._wrp_init_bounds.height * lim) {
+                    this.hide();
+                    return;
+                } else if (calc < 0)
+                    return;
+                this._wrapper_el.style.top = calc + 'px';
+                if (!this._wrapper_el.style.bottom) {
+                    this._wrapper_el.style.bottom = vh - 
+                        (this._wrp_init_bounds.y + this._wrp_init_bounds.height)
+                        + 'px';
+                }
+                this._wrapper_el.style.height =  "auto";
+            } else if (this._mouse_pos_mod == 'bottom') {
+                calc = vh - rect.y - rect.height - ydiff;
+                if ((rect.height + ydiff) < this._wrp_init_bounds.height * lim) {
+                    this.hide();
+                    return;
+                } else if (calc < 0)
+                    return;
+                this._wrapper_el.style.bottom = calc + 'px';
+                if (!this._wrapper_el.style.top)
+                    this._wrapper_el.style.top = this._wrp_init_bounds.y + 'px';
+                this._wrapper_el.style.height =  "auto";
+            }
+     }
+    }
+    private _mouseUpEvent() {
+        document.removeEventListener("mouseup", this._mouse_up_listener);
+        document.removeEventListener("touchend", this._mouse_up_listener);
+        document.removeEventListener("mousemove", this._mouse_move_listener);
+        document.removeEventListener("touchmove", this._mouse_move_listener);
+        this._mouse_move_listener = this._mouse_up_listener = null;
+    }
+    private _posStyles() {
+        let ret = []
+        let mc =[]
+        let dir = (s:any) => {
+            return `${Object.keys(s).map(k => `${k}:${s[k]}`).join(";")}`
+        }
+        for (let bp of this._position_breakpoints) {
+            this._change_mc = true;
+            if(bp.val.endsWith("center")) {
+                if(bp.val == 'center')
+                    mc.push(`@media screen and ${bp.cond} {.__modal-container 
+                           {align-items:flex-start;justify-content:center} 
+                           .__modal-wrapper {margin:auto;position:relative}}
+                    `);
+                else
+                    mc.push(`@media screen and ${bp.cond} 
+                           {.__modal-container {justify-content:center}}
+                    `);
+            }
```

Comment:
You can avoid creating more media definitions for the same condition and instead collapse all the conditional definitions in a variable and use it in only one media declaration.